opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -263,6 +299,15 @@ def get_aggregated_resources(
             detector = detectors[detector_ind]
             try:
                 detected_resources = future.result(timeout=timeout)
+                if (
+                    final_resource.schema_url != ""
+                    and detected_resources.schema_url != ""
+                    and final_resource != detected_resources
+                ):
+                    logger.error(
```

Comment:
You do not need to do the `schema_url` validations here. This will be handled in `merge`.

Reply:
I'm not sure how to handle the error in `get_aggregated_resource`. The detectors are responsible for setting the `schema_url` for the resources they detect, so I think it's appropriate to catch this error in this function instead of `merge`. However, when the resources are merged, they fail, and `merge` logs one or more errors that say nothing about the resource detector.

Reply:
Referring to my [comment](https://github.com/open-telemetry/opentelemetry-python/pull/1871#discussion_r640180544), if you include the `schema_urls` in question, it will identify which urls are conflicting, and in turn, the `ResourceDetector` responsible for them. I don't have a strong opinion on this though, since your way serves the same purpose.

Reply:
I think it's better to leave the error handling to `Resource.merge` now that the `schema_url`s are included in the logs. It's better to be verbose. I've made this change.